precompile/allow_list.go

```diff
@@ -11,23 +11,23 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
-// Singleton StatefulPrecompiledContracts for W/R access to the contract deployer allow list.
+// Singleton StatefulPrecompiledContract for W/R access to the contract deployer allow list.
 var (
-	ModifyAllowListPrecompile StatefulPrecompiledContract = &modifyAllowListPrecompile{}
-	ReadAllowListPrecompile   StatefulPrecompiledContract = &readAllowListPrecompile{}
+	AllowListPrecompile StatefulPrecompiledContract = createAllowListPrecompile()
+	_                   StatefulPrecompileConfig    = &AllowListConfig{}
 )
 
-type AllowListRole common.Hash
-
 // Enum constants for valid AllowListRole
 var (
-	None     AllowListRole = AllowListRole(common.Hash{})  // No role assigned - this is equivalent to common.Hash{} and deletes the key from the DB when set
-	Deployer AllowListRole = AllowListRole(common.Hash{1}) // Deployers are allowed to create new contracts
-	Admin    AllowListRole = AllowListRole(common.Hash{2}) // Admin - allowed to modify both the admin and deployer list as well as deploy contracts
-)
-
-const (
-	modifyAllowListInputLength = common.AddressLength + common.HashLength // Required length of an input to modify allow list precompile
+	None     common.Hash = common.BigToHash(big.NewInt(0)) // No role assigned - this is equivalent to common.Hash{} and deletes the key from the DB when set
```

Comment:
add `AllowListRole` type alias back in. Add AllowList prefix to these roles or move allow list into its own package.